### ISSUE TITLE
fantomas integration

### ISSUE THREAD

**Issue by yatli** [yatli/fsharp-language-server#8]
With fantomas we can have documentFormattingProvider and possibly range formatting capabilities.

**Comment by yatli** [comment]
Done.

**Comment by ErikFeeley** [comment]
@yatli  Hello!

I was about to create an issue for format on save but this came up in the suggestions :)

So I do have the fantomas tool installed globally. `dotnet tool list -g` shows
`fantomas-tool                      3.0.0-beta-002              fantomas`

and in my CocConfig I have the following
```
"coc.preferences.formatOnSaveFiletypes": [
        "css",
        "elm",
        "fsharp",
        "Markdown",
        "typescript",
        "typescriptreact",
        "json",
        "javascript",
        "javascriptreact",
        "rust"
    ],
```

However format on save is not triggering. 

After creating a simple console app and triggering a save in the generated Program.fs CocInfo shows the following.
```
Found 1 unused declarations in 45ms
Autocompleting at /home/erik/_dev/personal-repos/FsharpTemp/Console/Program.fs(15,9)
Trying to recompile Program.fs with timeout
Checked Program.fs in 180ms
Finished typecheck, looking for completions...
Cancelling request 1
Request 1 was cancelled
Checked Program.fs in 55ms
Found 1 unused declarations in 0ms
Checked Program.fs in 19ms
Found 1 unused declarations in 0ms
Found 1 unused declarations in 0ms

```

Sorry to bring this up in a closed issue, if it is more appropriate to create a new one please let me know thanks!

**Comment by yatli** [comment]
hi @ErikFeeley !

Have you tried `:call CocAction('format')`? We can first make sure that the fantomas integration actually works.

**Comment by ErikFeeley** [comment]
When i `:call CocAction('format')` nothing happens and i don't seem to get an error message. If i check `:CocInfo` after trying to call format there is nothing new reported.

**Comment by yatli** [comment]
try `:CocCommand fsharp.downloadLanguageServer` then -- the server is decoupled from the client and currently requires a manual update

**Comment by yatli** [comment]
I should really make an automatic updater :)

**Comment by ErikFeeley** [comment]
Sorry I took so long to get back tho this!

So i tried `:CocCommand fsharp.downloadLanguageServer` and i got this from a following `CocInfo` call.

```
[Error  - 8:07:44 PM] Starting client failed: 
Error: Command "/home/erik/.config/coc/extensions/coc-fsharp-data/server/FSharpLanguageServer" of fsharp is not executable: Error: not found: /home/erik/.config/coc/extensions/coc-fsharp-data/server/FSharpLanguageServer
    at LanguageClient.createMessageTransports (/home/erik/.local/share/nvim/plugged/coc.nvim/build/index.js:76455:23)
```

Just to be sure I am actually getting the updated server I also `rm -rf`-ed the server directory and let it download fresh again. Other functionality is working, however format on save is not. I also tried the coc-action to format selection which does not report any errors but also does nothing.

Perhaps my dotnet setup is causing an issue? I am using the 3.0 preview8 sdk, and I am also using the new tool manifest to specify local tools. But I also have fantomas-tool installed as a global tool as well for a fallback.

I have to take off, but I will re-visit next week!
